src/App.tsx

```diff
@@ -8,32 +10,46 @@ function getRandomName(): string {
 }
 
 export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
+  const [clockName, setTimerName] = useState('Clock-0');
+  const [hasClock, setClockOn] = useState(true);
+  const [prevClockName, setPrevName] = useState('Clock-0');
```

Comment:
Do you really need to store the previous name? You can get previous value as a parameter of setState 
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state